### PR TITLE
Issue1880 mover efficiency fix and validation rework

### LINKED ISSUE #1880: Correct mover efficiency in unusual configuration and rework validation models
Referred in #1879.
This issue is to address the two following items:

- There is an error in `Fluid.Movers.BaseClasses.FlowMachineInterface`. If the provided power curve refers to total power instead of hydraulic power (`powerOrEfficiencyIsHydraulic=false`), the motor efficiency $\eta_{mot}$ would be accounted for twice because of mistakes in equations. This configuration is rarely used in Buildings. Throughout the Buildings library, this error only affected results of `Fluid.Movers.Validation.{PowerEuler,PowerExact,PowerSimplified}`.
- There is an error in `Fluid.Movers.Validation.{PowerEuler,PowerExact}`. The efficiency values were assigned incorrectly. These two models were intended to demonstrate how the speed-controlled model and the flow-controlled models estimate power differently. However, there is no longer such difference once the efficiency assignments are corrected. After some email exchanges with @Mathadon, we think various refactoring over the years may have fixed the limitation these models tried to show.

These two are combined in the same ticket because they affect the same validation models.

Action items:
- [x] Correct errors in the efficiency computation of `Fluid.Movers.BaseClasses.FlowMachineInterface`.
- [x] Obsolete `Fluid.Movers.Validation.{PowerEuler,PowerExact,PowerSimplified}`.
- [x] Make a new validation model to demonstrate the difference of power estimation methods of the current version.
- [x] Make a new validation model to validate the efficiency computation that is corrected.

### PR DESCRIPTION
**Pull request by hcasperfu**
This closes #1880.

### CODE REVIEW

File: IBPSA/Fluid/Movers/BaseClasses/FlowMachineInterface.mo
Comment (mwetter) [inline IBPSA/Fluid/Movers/BaseClasses/FlowMachineInterface.mo:646]: Such a change in code needs an entry in the model's revision notes.

**Review by mwetter — CHANGES_REQUESTED** [review]
See inline comment for missing revision note. Otherwise it looks good (but the CI tests fail which needs to be corrected).

File: IBPSA/Fluid/Movers/BaseClasses/FlowMachineInterface.mo
Comment (hcasperfu) [reply IBPSA/Fluid/Movers/BaseClasses/FlowMachineInterface.mo:646]: Greetings, Michael. This PR is still a draft. I created it so that I don't check the todo boxes in the issue post out of thin air. I will add the change logs in due course.

**Comment by hcasperfu** [comment]
@mwetter - This PR is ready for review.
I obsoleted `Movers.Validation.{PowerExact,PowerEuler,PowerSimplified}` and made `Movers.Validation.{ComparePowerHydraulic,ComparePowerInput,ComparePowerTotal}`.

- `ComparePowerHydraulic` validates that $\eta = \eta_{hyd} \cdot \eta_{mot}$ under `per.powerOrEfficiencyIsHydraulic=true`.
- `ComparePowerTotal` validates the above under `per.powerOrEfficiencyIsHydraulic=true`.
- `ComparePowerInput` shows that the mover models with different inputs produce the same power estimation results.

I will work on a concurrent PR with Buildings.
Once you approve I will request an external review.

**Comment by hcasperfu** [comment]
In a70a9f5:
- The redeclarations in the validation models were rewritten to address an Optimica warning message.
- The changes caused the models to fail OpenModelica translation. OpenModelica fixed this glitch in https://github.com/OpenModelica/OpenModelica/issues/12484. These models are excluded in CI for now.

File: IBPSA/Fluid/Movers/BaseClasses/FlowMachineInterface.mo
Comment (Mathadon) [inline IBPSA/Fluid/Movers/BaseClasses/FlowMachineInterface.mo:616]: good catch

**Review by Mathadon — APPROVED** [review]
@hcasperfu this looks okay, thanks!

**Review by mwetter — APPROVED** [review]
This is ready to merge when the CI tests pass.

**Pull request merged**
